From 0d9222af51596690472d40a6af72661676c6a996 Mon Sep 17 00:00:00 2001
From: John Linn <john.linn@xilinx.com>
Date: Fri, 5 Mar 2010 14:07:05 -0700
Subject: [PATCH] Xilinx: Powerpc: added code to make gdb work

This patch was from Brian Hill and is needed to make gdb/gdbserver
work. I tried to get the mainline to take this code but they didn't
seem responsive, so it may stay in the xilinx tree.
---
 arch/powerpc/kernel/head_44x.S |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/kernel/head_44x.S b/arch/powerpc/kernel/head_44x.S
index 7dd2981..882461b 100644
--- a/arch/powerpc/kernel/head_44x.S
+++ b/arch/powerpc/kernel/head_44x.S
@@ -96,6 +96,18 @@ _ENTRY(_start);
 	bl	init_cpu_state
 
 	/*
+	 * The following code is needed to make GDB work with soft breakpoints.
+	 * This patch was provided by Brian Hill. 
+	 */ 
+	lis     r2,DBCR0_IDM@h
+	mtspr   SPRN_DBCR0,r2
+	isync
+
+	/* clear any residual debug events */
+	li      r2,-1
+	mtspr   SPRN_DBSR,r2
+
+	/*
 	 * This is where the main kernel code starts.
 	 */
 
-- 
1.7.0.4

